### ISSUE TITLE
AssemblyInfo (.NET Core) stamps .csproj files with empty XML attribute tags

### ISSUE THREAD

**Issue by jonathanpickard** [BMuuN/vsts-assemblyinfo-task#149]
When assigning attributes in the AssemblyInfo (.NET Core & Standard) configuration web page, the build pipeline runs successfully but the .csproj files have empty XML tags rather than having content

e.g.
    "Authors/>"
    "Company/>"
    "Product/>"
    "FileVersion/>"

We have discovered that the only way to get attributes set properly is to manually create a copy of the project file (with new filename) using Windows Explorer and then trigger the build again.  The "copied file" (which is not in the repository) gets modified correctly.  Of course this isn't a suitable workaround because it's not automated.  We've actually tried using a custom Powershell task to create the "copied file" within the build pipeline but the copied file ends up having empty attributes.  We cannot work out what the issue is.

**Comment by BMuuN** [comment]
Hey @jonathanpickard  any chance you post a copy of the task output here or ensure you have the file encoding set correctly.

**Comment by jonathanpickard** [comment]
Hi there.  Here's the task output (below).  We have played with all the file encoding option settings and none of them have made a positive difference in terms of getting AssemblyInfo to confirm that it has injected the information in the task parameters.

--

Author       : Bleddyn Richards
Help         : [More Information](https://marketplace.visualstudio.com/items?itemName=bleddynrichards.Assembly-Info-Task)
==============================================================================
Task Parameters...
	Source folder: d:\XSquadAgent2\_work\23\s
	Source files: **/*.csproj
	Insert attributes: true
	File encoding: auto
	Write unicode BOM: true
	Generate NuGet package on build: true
	Require license acceptance: false
	Package id: 
	Package version: 
	Authors: Jonathan Pickard
	Company: SD Worx Uk Ltd
	Product: A widget that prints free money
	Description: 
	Copyright: 
	License Url: 
	Project Url: 
	Icon Url: 
	Repository Url: 
	Repository type: 
	Tags: 
	Release notes: 
	Assembly neutral language: 
	Assembly version: 
	Assembly file version: 1.0.0.7536
	Informational version: 

Setting .Net Core assembly info...
Processing: d:\XSquadAgent2\_work\23\s\ROIPayrollSubmission.Api\ROIPayrollSubmission.Api.csproj
	Detected character encoding: iso-8859-1
	Verify character encoding: iso-8859-1

Processing: d:\XSquadAgent2\_work\23\s\ROIPayrollSubmission.Tests\ROIPayrollSubmission.Tests.csproj
	Detected character encoding: iso-8859-1
	Verify character encoding: iso-8859-1

Complete.
Finishing: Set Assembly Manifest Data

**Comment by BMuuN** [comment]
Are you able to post the contents of either the `ROIPayrollSubmission.Api.csproj` or `ROIPayrollSubmission.Tests.csproj` file?

**Comment by jonathanpickard** [comment]
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup Label="Globals">
    <SccProjectName>SAK</SccProjectName>
    <SccProvider>SAK</SccProvider>
    <SccAuxPath>SAK</SccAuxPath>
    <SccLocalPath>SAK</SccLocalPath>
  </PropertyGroup>
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <AspNetCoreHostingModel>OutOfProcess</AspNetCoreHostingModel>
    <GeneratePackageOnBuild/>
    <PackageRequireLicenseAcceptance/>
    <Authors/>
    <Company/>
    <Product/>
    <FileVersion/>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="IdentityServer4" Version="2.5.2"/>
    <PackageReference Include="IdentityServer4.AccessTokenValidation" Version="2.7.0"/>
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.5"/>
    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.5"/>
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.2.3"/>
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3"/>
    <PackageReference Include="Serilog.AspNetCore" Version="3.2.0"/>
    <PackageReference Include="Serilog.Sinks.RollingFile" Version="3.3.0"/>
  </ItemGroup>
</Project>

**Comment by jonathanpickard** [comment]
![image](https://user-images.githubusercontent.com/40670641/95838508-c091a700-0d39-11eb-922e-51e388727e2f.png)

**Comment by jonathanpickard** [comment]
![image](https://user-images.githubusercontent.com/40670641/95838642-ec149180-0d39-11eb-8329-f0cadecd4d60.png)

**Comment by BMuuN** [comment]
This one has me puzzled!  All my local testing has your scenario working 100% on my local dev machine which I know is no help to you but I can't see why it's not working for you.  Is the project available in GitHub or are you able to post the `*.csproj` files as `*.txt` files?

**Comment by jonathanpickard** [comment]
[ROIPayrollSubmission.Apiutf8.csproj.txt](https://github.com/BMuuN/vsts-assemblyinfo-task/files/5598202/ROIPayrollSubmission.Apiutf8.csproj.txt)
[ROIPayrollSubmission.Tests.csproj.txt](https://github.com/BMuuN/vsts-assemblyinfo-task/files/5598203/ROIPayrollSubmission.Tests.csproj.txt)
[ROIPayrollSubmission.Api.csproj.txt](https://github.com/BMuuN/vsts-assemblyinfo-task/files/5598204/ROIPayrollSubmission.Api.csproj.txt)
[ROIPayrollSubmission.ApiCopy.csproj.txt](https://github.com/BMuuN/vsts-assemblyinfo-task/files/5598205/ROIPayrollSubmission.ApiCopy.csproj.txt)

**Comment by jonathanpickard** [comment]
Only thing I can think of that is stopping this from working is that our build agent server is not running your add in properly - maybe something to do with permissions maybe?

**Comment by tallalkazmi** [comment]
I'm facing the same issue, is there any solution to this?

**Comment by tallalkazmi** [comment]
> I'm facing the same issue, is there any solution to this?

I figured it out. I just checked the box saying **Insert attributes** and it works perfectly now.

![image](https://user-images.githubusercontent.com/31439236/108626385-47cdf480-7469-11eb-8465-718aa88dfee3.png)

**Comment by jonathanpickard** [comment]
@tallalkazmi your solution does indeed insert tags in to the respective C# project files, but what I'm seeing is that certain tags are empty.  This is confirmed when I look at the Azure DevOps Build Pipeline: it tells me the task parameters (I'm setting quite a few!) but when the component comes to apply the task parameters to the matching C# project files, the component confirms that only two of those parameters are being set.  It's as if the component doesn't want to apply anything to do with version numbers.

Here's the output of my task:

Starting: Set Assembly Manifest Data
==============================================================================
Task         : Assembly Info (.Net Core & .Net Standard)
Description  : Set assembly or package information such as Version, Copyright and Repository URL's etc.
Version      : 2.0.67
Author       : Bleddyn Richards
Help         : [More Information](https://marketplace.visualstudio.com/items?itemName=bleddynrichards.Assembly-Info-Task)
==============================================================================
Task Parameters...
	Source folder: d:\Agent2\_work\29\s
	Source files: **/*.csproj
	Insert attributes: true
	File encoding: auto
	Write unicode BOM: false
	Generate NuGet package on build: false
	Require license acceptance: false
	Package id: 
	Package version: 8.7.6.5
	Authors: Jonathan Pickard
	Company: 
	Product: An API
	Description: 
	Copyright: (c) 2021
	License Url: 
	Project Url: 
	Icon Url: 
	Repository Url: 
	Repository type: 
	Tags: 
	Release notes: 
	Assembly neutral language: 
	Assembly version: 5.6.7.8
	Assembly file version: 1.2.3.4
	Informational version: 3.5.7.9

Setting .Net Core assembly info...
Processing: d:\Agent2\_work\29\s\PaymentFileDecryptClient\PaymentFileDecryptClient.csproj
	Detected character encoding: iso-8859-2
	Authors --> Jonathan Pickard
	Copyright --> (c) 2021
	Verify character encoding: iso-8859-2

Processing: d:\Agent2\_work\29\s\ROIPayrollPayment.Api.Tests\ROIPayrollPayment.Api.Tests.csproj
	Detected character encoding: iso-8859-1
	Detected character encoding is different to the one specified.
	Verify character encoding: iso-8859-1

**Comment by BMuuN** [comment]
@jonathanpickard apologies for the late reply.  If this is still an issue for you could you try setting the following:
```
File encoding: 'utf-8'
Write unicode BOM: true
```

**Comment by BMuuN** [comment]
Closing due to no response.  Please re-open the issue if you are still facing the issue on v3 of the extension.

**Comment by jonathanpickard** [comment]
Hi there - I've only just around to having another attempt at testing this.
Couple of things:
1) I can't see version 3 in the version drop down menu in DevOps.  Only version 2 seems to be available.  Have you released the component under a different name or is version 3 under the existing name - not available yet?
2) What version of Node.exe does your software work with?  When the build pipeline is running (with the diagnostics checkbox checked) I can see your task picking up Node version 6.17.1.0

So in summary, I suspect the problem is the server that the build agent is running on.  We have 5 build agents on the single virtual server and any of these 5 build agents (which seem to be picked at random when running a new build) all end up with the same result - a csproj file with tags that have no values.

**Comment by jonathanpickard** [comment]
Hi again,

You might not believe this, but we are using the old Market Place extension of yours.  We have not yet installed the latest Market Place extension of yours (one that noticeably has a different icon)!

I've requested our Admins to install the latest component and then I'll see if our problem goes away.

Thanks.

**Comment by jonathanpickard** [comment]
Hello Bleddyn, great news - we are using your latest Marketplace extension and we are now seeing assembly info in the DLLs that we have built.  Thank you for all your help.
